ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/IAuditAuthenticator.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Telefonaktiebolaget LM Ericsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecaudit.auth;
+
+import java.util.Optional;
+
+import org.apache.cassandra.auth.IAuthenticator;
+
+/**
+ * An {@link IAuthenticator} that provide custom fields to be used during authentication auditing.
+ */
+public interface IAuditAuthenticator extends IAuthenticator, IOptionsProvider
+{
+    /**
+     * Returns an <em>audited</em> {@link AuditSaslNegotiator SaslNegotiator}.
+     * <p>
+     * Note: Implementations should probably use this method when {@link IAuthenticator#newSaslNegotiator()} is called.
+     *
+     * @return an instance of an {@link AuditSaslNegotiator}
+     */
+    AuditSaslNegotiator newAuditSaslNegotiator();
```

Comment:
One somewhat annoying thing is that in more resent versions of Cassandra the original newSaslNegotiator() method is accepting an InetAddress as parameter. So I guess we're choosing between 1) making this interface different between ecAudit flavors, or 2) adding the parameter in all flavors but sometimes accepting it as null (since there is no way to inject useful information here).

I leaning towards option 1. What are  your thoughts? Is there a clever option 3 that I'm overlooking?

Reply:
I don't think it makes sense to add parameters that we aren't using in one flavor; the flavors are different because Cassandra is different between majors versions so I think it's fine.

Reply:
Yeah Let's leave it like this, and let's remember to update this as we merge forward in flavors.